aten/src/ATen/core/functional.h

```diff
@@ -34,6 +43,18 @@ inline std::vector<R> fmap(const T& inputs) {
   return r;
 }
 
+template<class F, class InputIt, class OutputIt>
+inline void fmap_to(InputIt first, InputIt last, OutputIt d_first, const F& fn) {
+  while (first != last) {
+    *d_first++ = *first++;
+  }
+}
+
+template<class F, class From, class To>
+inline void fmap_to(const From& inputs, To& outputs, const F& fn) {
+  return fmap_to(inputs.begin(), inputs.end(), std::back_inserter(outputs), fn);
```

Comment:
Nit:
Reserve space in `outputs` before calling `fmap_to` to save memory allocation time

Reply:
I wanted to do this, but for some reason I couldn't get `std::distance` to work in places where `fmap` is used with a `ListIterator`. I get that `operator-` is not implemented, even though I can look in `List.h` and see it there as a public method! I also verified that `begin` and `end` are implemented for `List`.

Reply:
The issues may have been coming from `ListIterator` in `tree_views.h`. That being said, I the problems we've had here show that `std::distance` is not necessarily a universal solution. I'm settling for allowing the user to optionally pass in a size. I'm keeping the `reserve` in the original `fmap`, so we shouldn't see any perf loss